### ISSUE TITLE
Fix webpack 5 deprecation warnings

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#15947]
Next.js fully supports webpack 5, however, some of our plugins are using deprecated APIs.

Before we can consider webpack 5 stable, we should completely eliminate deprecated webpack 5 API usage.

Known warnings:

- [x] https://github.com/webpack-contrib/terser-webpack-plugin/pull/283
- [ ] https://github.com/webpack-contrib/mini-css-extract-plugin (most warnings come from this plugin)
- [x] Next.js internal webpack plugins warnings
  - [x] https://github.com/vercel/next.js/pull/15797
  - [x] https://github.com/vercel/next.js/pull/15806

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
